python/src/ir.cc

```diff
@@ -1575,6 +1575,19 @@ void init_triton_ir(py::module &&m) {
         if (triton::tools::getBoolEnv("TRITON_ENABLE_LLVM_DEBUG")) {
           ::llvm::DebugFlag = true;
         }
+
+        auto *context = self.getContext();
+        bool haveDiagnostics =
```

Comment:
nit: combine this into the if-statement

Reply:
I thought the "run" path should always be preceded with the "enable_debug" call which handles `MLIR_ENABLE_DIAGNOSTICS`. Was your case where  `enable_debug` was not called?